### ISSUE TITLE
Relaunch Forgetful: Positions, Sizes, Which Windows were Open in Last Session

### ISSUE THREAD

**Issue by washere** [tomboy-notes/tomboy-ng#92]
On relaunch of the app, it should remember the following just like the Tomboy original:

**________________________________________________________**

1) Which Notes Were open/visible in last session:
open the same notes that were open at the end of last session
**________________________________________________________**

2) Positions of the Notes:
Remember the position of the notes that were open/visible on the screen
This makes Tomboy original act like a corkboard
But in ng the positions are forgotten on relaunch
**________________________________________________________**

3) Sizes of Notes & Search:
In Tomboy original the sizes of the Notes panels are remembered from end of last session on relaunch
For example for me Some are thin (lists), some are wider (chunks of text), some are small, some are big etc

Also the same for the SEARCH PANEL, it's size AND remembered within the same session (closing and re-opening) but not after quitting and relaunching the app.

Search panel position should also be remembered as in above point 2, currently even within the same session the position of re-opened Search panel is lost.
Note panels are worse, re-opened notes within the same session lose their position and size even within the same session after closing & re-opening

**________________________________________________________**

Basically every note AND the Search panel need to have under the bonnet:

* Coordinates for their position from last session or when they were closed (Top Left X,Y)
* Coordinates for their size (Top Left X,Y + Bottom Right X,Y)

thanks agian for this app.

**Comment by davidbannon** [comment]
Discussion mode -
I'm going to have to work out just how Tomboy does this. If it chooses to rewrite every open note at shutdown time, that could be a significant delay to shutdowns (only for many, large notes). OS builders are a lot less tolerant these days of apps that don't want to shutdown quickly when told to do so. I guess its time I stopped trying to maintain strict Tomboy compatibility so, perhaps, I could write a out a restart file listing notes that were open at previous shutdown, if the OS kills me before that's finished it would not involve any data loss (this is Windows only issue, Linux users are probably using a journaled filesystem, mac ?). And I would restore the note to the position it was at when last saved (as the tomboy xml format provides), not necessarily when shutdown occurred. For notes being actively written to that would produce the same result but for a note only being read and moved around the desktop may not be want user wants. But better that than delaying shutdown.
Good little project while sitting in an aeroplane for far too long ...
Davo

**Comment by washere** [comment]
Yes a little working cache file with:

1) Names of open panels (notes and-if-also search)

2) Position (top-left X,Y coordinates)

3) Sizes of ALL windows, whether open or closed (top-left + bottom-right coordinates)

This last one is actually the most important for serious users with many Windows.

Thanks again.

**Comment by washere** [comment]
Interestingly Microsoft fixed both issues I referenced here, persistence relaunch in remembering the sizes & positions of notes in their app:

https://mspoweruser.com/microsoft-pushes-an-update-to-both-of-their-note-taking-tools-in-windows-10/amp/

**Comment by davidbannon** [comment]
Well, not sure when Microsoft fixed their's but I've had -ng doing all of that for a couple of weeks now :-)
So it will be all good next release (or now if you want to build yourself).  I am currently overseas and running a pretty tight schedule so not getting much done..... Will be back home for a couple of weeks in June and hope to get enough testing done to push out a release during that period.
Davo

**Comment by washere** [comment]
Thanks Davo. Most of us are patient and appreciative. So I will wait for the release when you get back, no hurry anyway. Also can't wait to finally get the Dark Mode on windows. Just updated (force update) to Windows 1903 big May update and still no dark mode inherited from MS Win Dark theme settings. 

I have other simple UI suggestions to make ng even cooler but will wait till after dark mode. 

Have fun and bon voyage. 
Thanks again.

**Comment by davidbannon** [comment]
OK, v0.23 now does this.
